app/services/piece-access-level-service.js

```diff
@@ -62,4 +66,32 @@ export default class PieceAccessLevelService extends Service {
       previousPiece = await previousPiece.previousPiece;
     }
   }
+
+  /*
+   * This method is used to update the access levels of the pieces of a subcase's treatments. When a subcase has
+   * been updated to being confidential, all the pieces connected to it through agenda-item-treatments get the
+   * "Ministerraad" access level.
+   */
+  async updateDecisionsAccessLevelOfSubcase(subcase) {
+    await subcase.preEditOrSaveCheck();
```

Comment:
I'm not familiar with the `preEditOrSaveCheck()`. What's the specific reason to call it here?

Reply:
`preEditOrSaveCheck` is the concurrency check on agendaitem/subcase/newsletter-info model.
Typically you use this if you risk saving the model with stale attributes or serialized relations and will block model save until you refresh the page.